src/components/SignInSteps/TermsStep.tsx

```diff
@@ -53,7 +55,11 @@ export const TermsStep: React.FC = () => {
         </TermsOverlay>
       </TermsBox>
       <StepFooter>
-        <ContinueButton to={absoluteRoutes.studio.newMembership()} disabled={!hasScrolledToBottom}>
+        <ContinueButton
+          onClick={decrementOverlaysOpenCount}
```

Comment:
I don't like this very much, it leaks overlay implementation into a view that doesn't have anything to do with it. What if instead we made sure the `BaseDialog` will call this whatever happens?

One idea I have in mind is we could try the to keep the cleanup `useEffect` we have there but only make the call if the transition haven't fired? That should keep us safe, what do you think?

Reply:
> One idea I have in mind is we could try the to keep the cleanup `useEffect` we have there but only make the call if the transition haven't fired? That should keep us safe, what do you think?

Tried that earlier with no success. Do you have an idea how we could implement that? 

I've tried 
```js
// BaseDialog.tsx
const isOnExitCalled = useRef(false)
// ...
useEffect(() => {
    if (showDialog && !isOnExitCalled.current) {
      return () => {
        decrementOverlaysOpenCount()
      }
    }
  }, [decrementOverlaysOpenCount, showDialog])
//... 

      <CSSTransition
        in={showDialog}
        timeout={200}
        classNames={transitions.names.dialog}
        mountOnEnter
        unmountOnExit
        onExit={() => {
          isOnExitCalled.current = true
        }}
        onExited={() => {
          if (isOnExitCalled.current) {
            decrementOverlaysOpenCount()
          }
        }}
      >
```
and something analogous with using `useState`. If I close the dialog by clicking the `close button`, the decrementOverlaysOpenCount is called two times(from `onExited` and from the cleanup function).